### PR TITLE
Feature/add link to person dynamics

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

This is to link address, name, identifier to person entity in dynamics.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested? 
Tested locally.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: FAMS-1398

### CODE REVIEW

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/Person/SSG_Person.cs [hunk 1]
```diff
@@ -6,14 +6,17 @@
 
 namespace Fams3Adapter.Dynamics.Person
 {
-    public class SSG_Person 
+    //if we add PersonId in this class, when we do odataClient insert, it will fail with message personID cannot be null.
+    //so, we have to move PersonId to SSG_Person class. When we do insert request, there is no personID (as we use SSG_Person_Upload)
+    //and when we get the result, it has personId (as SSG_Person has personId.) 
+    public class SSG_Person_Upload
```
Comment (rantifamilusi) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/Person/SSG_Person.cs]: Can we change name to PersonEntity

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonFoundService.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DynamicsAdapter.Web.PersonSearch
+{
+    public interface IPersonFoundService
```
Comment (rantifamilusi) [inline app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonFoundService.cs]: Change name to ISearchResultService .

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonFoundService.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    {
+        Task<bool> ProcessPersonFound(Person person, ProviderProfile providerProfile, SSG_SearchRequest searchRequest, CancellationToken cancellationToken);
+    }
+
+    public class PersonFoundService : IPersonFoundService
```
Comment (rantifamilusi) [inline app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonFoundService.cs]: Change Name to SearchResultService

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonFoundService.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonFoundService.cs]: ok

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonFoundService.cs (hunk 3 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonFoundService.cs]: ok

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/Person/SSG_Person.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/Fams3Adapter.Dynamics/Person/SSG_Person.cs]: ok

**Pull request merged**